scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3852,448 @@ def rvs(self, dim, size=1, random_state=None):
+            raise ValueError(f"Size of each quantile should be size of 'm': "
+                             f"received {xx.shape[-1]}, "
+                             f"but expected {m.shape[-1]}.")
+
+        xcond = np.any(xx != x, axis=-1)
+        xcond = xcond | np.any((xx < 0) | (xx > m), axis=-1)
+        xcond = xcond | (xx.sum(axis=-1) != n)
+        return xx, xcond
+
+    def _checkresult(self, result, cond, bad_value):
+        result = np.asarray(result)
+        if cond.ndim != 0:
+            result[cond] = bad_value
+        elif cond:
+            return bad_value
+        if result.ndim == 0:
+            return result[()]
+        return result
+
+    def _logpmf(self, x, M, m, n):
+        num = betaln(m+1, 1) - betaln(x+1, m-x+1)
+        den = betaln(M+1, 1) - betaln(n+1, M-n+1)
+        return num.sum(axis=-1) - den
+
+    def logpmf(self, x, m, n):
+        """
+        Log of the Multivariate Hypergeometric probability mass function.
+
+        Parameters
+        ----------
+        x : array_like
+            Quantiles, with the last axis of `x` denoting the components.
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        logpmf : ndarray or scalar
+            Log of the probability mass function evaluated at `x`
+
+        Notes
+        -----
+        %(_doc_callparams_note)s
+        """
+        M, m, n, mncond = self._process_parameters(m, n)
+        x, xcond = self._process_quantiles(x, M, m, n)
+
+        result = self._logpmf(x, M, m, n)
+
+        # replace values for which x was out of the domain; broadcast
+        # xcond to the right shape
+        xcond_ = xcond | np.zeros(mncond.shape, dtype=np.bool_)
+        result = self._checkresult(result, xcond_, np.NINF)
+
+        # replace values bad for n or m; broadcast
+        # mncond to the right shape
+        mncond_ = mncond | np.zeros(xcond.shape, dtype=np.bool_)
+        return self._checkresult(result, mncond_, np.NAN)
+
+    def pmf(self, x, m, n):
+        """
+        Multivariate Hypergeometric probability mass function.
+
+        Parameters
+        ----------
+        x : array_like
+            Quantiles, with the last axis of `x` denoting the components.
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        pmf : ndarray or scalar
+            Probability density function evaluated at `x`
+
+        Notes
+        -----
+        %(_doc_callparams_note)s
+        """
+        out = np.exp(self.logpmf(x, m, n))
+        return out
+
+    def mean(self, m, n):
+        """
+        Mean of the Multivariate Hypergeometric distribution
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        mean : array_like or scalar
+            The mean of the distribution
+        """
+        M, m, n, mncond = self._process_parameters(m, n)
+        M, n = M[..., np.newaxis], n[..., np.newaxis]
```

Comment:
This is necessary because, at any instance, the dimensions of `M` and `n` must be aligned off the last axis of `m` otherwise numpy will misinterpret giving wrong results. Here's what I mean:

If we give arguments, `m=[10, 20]`, `n=[5, 6]`, what we mean is, we want to evaluate the mean value for two independent populations, one with sample size `5` and other with sample size `6`. This needs the shape of `n` to be `(2, 1)` so that `m` broadcasts to the shape `(2, 1)` giving us two independent populations. Otherwise, `n` will be wrongly interpreted (meaning, `n` will be aligned with two different `m` values) giving wrong result.

Reply:
The same argument goes for `var` and `cov` methods

Reply:
I think I see: each row of `m` represent a different "population", so for each row of `m`, there should be a corresponding row in `n`. However, the user actually provides `n` as a single row, like:
```res = multivariate_hypergeom.pmf(x=[[1, 11], [1, 12], [1, 13]], m=[[10, 20], [10, 30], [10, 40]], n=[12, 13, 14])```
so need to make it a column.

What if the user provides `n` as a column? Isn't this really what they _should_ be doing? If the user expects broadcasting to work with basic NumPy operations, they need to align their arrays properly. Why shouldn't they have to here?
```
res = multivariate_hypergeom.pmf(x=[[1, 11], [1, 12], [1, 13]], m=[[10, 20], [10, 30], [10, 40]], n=[[12], [13], [14]])
```
produces a matrix. Isn't that weird?

Reply:
This brings to mind that x, m, and n must be broadcastable after an extra dimension is added to the end of n, right?

Reply:
> This brings to mind that x, m, and n must be broadcastable after an extra dimension is added to the end of n, right?

Yes. They must be broadcastable once the extra dimensions have been added.

> there should probably be a check for this up front so that the user gets a sensible error message rather than one produced deep within the code.

Having that would be good. But IMO, it's not _really_ necessary as the exception raised without it will contain enough information to debug it (as numpy will raise the same exception under tho hood). At least, other parts of the code in scipy never check or broadcast explicitly but leave that to numpy. What do you say?

Reply:
Hmm, it's pretty common to check the dimensions of arrays for compatibility before entering the routine. But I can add it if you disagree.

Reply:
> it's pretty common to check the dimensions of arrays for compatibility before entering the routine

Ah, alright! I will do it then

Reply:
This is also done! All the arrays are broadcast first in the `_process_parameters` and `_process_quantiles` methods. If they are not broadcastable, it will generate error right there instead of deep within the code.